libcudacxx/include/cuda/std/detail/libcxx/include/__iterator/move_iterator.h

```diff
@@ -114,15 +114,15 @@ class _LIBCUDACXX_TEMPLATE_VIS move_iterator
   _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY static constexpr auto
   __mi_get_iter_concept()
   {
-    if constexpr (random_access_iterator<_Iter>)
+    if _LIBCUDACXX_CONSTEXPR_IF (random_access_iterator<_Iter>)
```

Comment:
This needs the MSVC2017 workaround rather than this

Reply:
What is that?

Reply:
I defined the macro this way: 

```
#if _LIBCUDACXX_STD_VER > 14 && !defined (_LIBCUDACXX_HAS_NO_CXX17_CONSTEXPR_IF)
#  define _LIBCUDACXX_CONSTEXPR_IF constexpr
#else
#  define _LIBCUDACXX_CONSTEXPR_IF
#endif
```

Reply:
This wont compile, because the different code paths have different return types. You need to use the MSVC_2017 workaround which does not define the function but uses a nested conditional.